modules/siddhi-extensions/markov-models/src/test/java/org/wso2/siddhi/extension/markov/MarkovChainTestcase.java

```diff
@@ -0,0 +1,291 @@
+                        Assert.assertEquals(0.0, event.getData(3));
+                        Assert.assertEquals(false, event.getData(4));
+                        break;
+                    case 3:
+                        Assert.assertEquals(0.3, event.getData(3));
+                        Assert.assertEquals(false, event.getData(4));
+                        break;
+                    case 4:
+                        Assert.assertEquals(0.0, event.getData(3));
+                        Assert.assertEquals(false, event.getData(4));
+                        break;
+                    case 5:
+                        Assert.assertEquals(0.6, event.getData(3));
+                        Assert.assertEquals(false, event.getData(4));
+
+                        break;
+                    case 6:
+                        Assert.assertEquals(0.6000000000000001, event.getData(3));
+                        Assert.assertEquals(false, event.getData(4));
+                        break;
+                    case 7:
+                        Assert.assertEquals(0.6, event.getData(3));
+                        Assert.assertEquals(false, event.getData(4));
+                        break;
+                    case 8:
+                        Assert.assertEquals(0.3, event.getData(3));
+                        Assert.assertEquals(false, event.getData(4));
+                        break;
+                    case 9:
+                        Assert.assertEquals(0.3, event.getData(3));
+                        Assert.assertEquals(false, event.getData(4));
+                        break;
+                    case 10:
+                        Assert.assertEquals(0.0, event.getData(3));
+                        Assert.assertEquals(false, event.getData(4));
+                        break;
+                    case 11:
+                        Assert.assertEquals(0.3, event.getData(3));
+                        Assert.assertEquals(false, event.getData(4));
+                        break;
+                    default:
+                        Assert.fail();
+                    }
+                }
+            }
+        });
+
+        InputHandler inputHandler = executionPlanRuntime.getInputHandler("InputStream");
+        executionPlanRuntime.start();
+
+        inputHandler.send(new Object[] { "1", "state01" });
+        inputHandler.send(new Object[] { "2", "state02" });
+        inputHandler.send(new Object[] { "1", "state03" });
+        inputHandler.send(new Object[] { "3", "state01" });
+        inputHandler.send(new Object[] { "3", "state02" });
+        inputHandler.send(new Object[] { "1", "state01" });
+        inputHandler.send(new Object[] { "1", "state02" });
+        inputHandler.send(new Object[] { "2", "state01" });
+        inputHandler.send(new Object[] { "2", "state03" });
+        inputHandler.send(new Object[] { "4", "state01" });
+        inputHandler.send(new Object[] { "4", "state03" });
+
+        Thread.sleep(100);
```

Comment:
Thanks for the suggestion. I changed it to a common approach used in siddhi. Now it will stop waiting when the expected no of events have arrived. Fixed with https://github.com/wso2/siddhi/pull/189/commits/690b60924a459f505cce35f57d2ce6b05e4980c2